### PR TITLE
VPA - Adding min and max resource allowed e2e tests

### LINKED ISSUE #2784: Add VPA e2e tests for recommender respecting Min and MaxAllowed
VPA Spec allows to specify minimum and maximum amount that can be recommended:
https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go#L206

We should have a test checking that these constraints are not violated.
@krzysied

### PR DESCRIPTION
**Pull request by krzysied**
/fixes #2784

### CODE REVIEW

File: vertical-pod-autoscaler/e2e/v1/recommender.go [hunk 1]
```diff
@@ -238,6 +238,85 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
...
+			apiv1.ResourceCPU: ParseQuantityOrDie(fmt.Sprintf("%dm", minMilliCpu)),
+		}
+		vpaCRD := createVpaCRDWithMinMaxAllowed(f, minAllowed, nil)
+
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(vpa.Status.Recommendation.ContainerRecommendations).Should(gomega.HaveLen(1))
+		cpu := getMilliCpu(vpa.Status.Recommendation.ContainerRecommendations[0].Target)
+		gomega.Expect(cpu).Should(gomega.BeNumerically(">=", minMilliCpu))
```
Comment (bskiba) [inline vertical-pod-autoscaler/e2e/v1/recommender.go]: Can you add logs to all the expects for the recommendation values to say what we are comparing? It makes debugging failed tests way easier

File: vertical-pod-autoscaler/e2e/v1/recommender.go [hunk 2]
```diff
@@ -238,6 +238,85 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
...
+		gomega.Expect(cpuUncapped).Should(gomega.BeNumerically(">", maxMilliCpu))
+	})
+})
+
+func getMilliCpu(resources apiv1.ResourceList) int64 {
+	cpu := resources[apiv1.ResourceCPU]
+	return cpu.MilliValue()
+}
+
+// createVpaCRDWithMinMaxAllowed creates vpa object with container policy
```
Comment (bskiba) [inline vertical-pod-autoscaler/e2e/v1/recommender.go]: createVpaCRDWithMinMaxAllowed creates vpa object with min an max resources allowed

File: vertical-pod-autoscaler/e2e/v1/recommender.go (hunk 1 above)
Comment (krzysied) [reply vertical-pod-autoscaler/e2e/v1/recommender.go]: Done.

File: vertical-pod-autoscaler/e2e/v1/recommender.go (hunk 2 above)
Comment (krzysied) [reply vertical-pod-autoscaler/e2e/v1/recommender.go]: Done.

File: vertical-pod-autoscaler/e2e/v1/recommender.go [hunk 3]
```diff
@@ -238,6 +238,88 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
...
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(vpa.Status.Recommendation.ContainerRecommendations).Should(gomega.HaveLen(1))
+		cpu := getMilliCpu(vpa.Status.Recommendation.ContainerRecommendations[0].Target)
+		gomega.Expect(cpu).Should(gomega.BeNumerically(">=", minMilliCpu),
+			fmt.Sprintf("target cpu recommendation should be greater than or equal to %dm", minMilliCpu))
+		cpuUncapped := getMilliCpu(vpa.Status.Recommendation.ContainerRecommendations[0].UncappedTarget)
+		gomega.Expect(cpuUncapped).Should(gomega.BeNumerically("<", minMilliCpu),
+			fmt.Sprintf("uncapped target cpu recommendation should be less then %dm", minMilliCpu))
```
Comment (bskiba) [inline vertical-pod-autoscaler/e2e/v1/recommender.go]: s/then/than/

File: vertical-pod-autoscaler/e2e/v1/recommender.go [hunk 4]
```diff
@@ -238,6 +238,88 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
...
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(vpa.Status.Recommendation.ContainerRecommendations).Should(gomega.HaveLen(1))
+		cpu := getMilliCpu(vpa.Status.Recommendation.ContainerRecommendations[0].Target)
+		gomega.Expect(cpu).Should(gomega.BeNumerically("<=", maxMilliCpu),
+			fmt.Sprintf("target cpu recommendation should be less than or equal to %dm", maxMilliCpu))
+		cpuUncapped := getMilliCpu(vpa.Status.Recommendation.ContainerRecommendations[0].UncappedTarget)
+		gomega.Expect(cpuUncapped).Should(gomega.BeNumerically(">", maxMilliCpu),
+			fmt.Sprintf("uncapped target cpu recommendation should be greater then %dm", maxMilliCpu))
```
Comment (bskiba) [inline vertical-pod-autoscaler/e2e/v1/recommender.go]: Same here

File: vertical-pod-autoscaler/e2e/v1/recommender.go (hunk 3 above)
Comment (krzysied) [reply vertical-pod-autoscaler/e2e/v1/recommender.go]: Done.

File: vertical-pod-autoscaler/e2e/v1/recommender.go (hunk 4 above)
Comment (krzysied) [reply vertical-pod-autoscaler/e2e/v1/recommender.go]: Done.

**Pull request merged**